plugins/synaptics-rmi/fu-synaptics-rmi-v7-device.c

```diff
@@ -326,8 +321,58 @@ fu_synaptics_rmi_v7_device_write_partition(FuSynapticsRmiDevice *self,
 							     fu_chunk_get_data_sz(chk),
 							     error))
 			return FALSE;
-		fu_progress_step_done(progress);
 	}
+
+	/* write signature */
+	if (g_ptr_array_index(array_img, 1)) {
+		fu_byte_array_append_uint16(req_offset, 0x0, G_LITTLE_ENDIAN);
+		if (!fu_synaptics_rmi_device_write(self,
+						   f34->data_base + 0x2,
+						   req_offset,
+						   FU_SYNAPTICS_RMI_DEVICE_FLAG_NONE,
+						   error)) {
+			g_prefix_error(error, "failed to write offset: ");
+			return FALSE;
+		}
+
+		chunks = fu_chunk_array_new_from_bytes(g_ptr_array_index(array_img, 1),
```

Comment:
yup, you're leaking memory -- splitting this out will allow you to define your "own" chunks.